pslr_model.c

```diff
@@ -778,12 +780,16 @@ static
 char *read_json_file(int *jsonsize) {
     int jsonfd = open("pentax_settings.json", O_RDONLY);
     if (jsonfd == -1) {
+#ifndef RAD10
         // cannot find in the current directory, also checking PKTDATADIR
         jsonfd = open(PKTDATADIR "/pentax_settings.json", O_RDONLY);
         if (jsonfd == -1) {
+#endif
             fprintf(stderr, "Cannot open pentax_settings.json file\n");
             return NULL;
+#ifndef RAD10
         }
+#endif
```

Comment:
Why do we need this special handling? 

If we really need to avoid checking the PKDATADIR directory, then it would be better to define a variable for this at the beginning of the file, set it to false for RAD10 and true for others and you that variable here. We should not have ifdef RAD10s in the middle of the code.

Reply:
PKTDATADIR is defined in the Makefile as:
PREFIX ?= /usr/local
PKTDATADIR = $(PREFIX)/share/pktriggercord

That seems useless to me in Windows. I could change the ifdef to ifdef PKTDATADIR, because it describes the test better.
And Yes I agree about not having Ifdefs in the middle of the code.

Reply:
I had 2 problems with this.

1. I dont really understand what PKTDATADIR stands for.
2. In order to compile the PKTDATADIR needs to be defined. But what should it point to? In Windows it is very common that the executable gets installed to something the user chooses.

The solution I am proposing is this:
- I define PKTDATADIR as ""
- In code I check if the directory exists. If not I skip the open.
- Other platforms can have PKTDATADIR defined as before, and should not be affected.

Reply:
PKTDATADIR is the directory where we store a few data files for pktriggercord. For instance pentax_settings.json

In Linux usually the binary is stored in /usr/bin and these files are stored in e.g. /usr/share/pktriggercord

It is defined in the `Makefile`

In the code when we look for a datafile usually we check the current directory and PKTDATADIR as well.

In the windows cross-compilation it's defined as "." which is the same as the current directory we check anyway. It was working so far. Maybe you could also define it "." but it's not that important.

I like the dir_exists change, but I have to test it under Linux.

Reply:
Looks good.